src/eureka/S3_data_reduction/wfc3.py

```diff
@@ -613,130 +806,143 @@ def fit_bg(dataim, datamask, datav0, datavariance, n, meta, isplots=0):
-        # for observations with only one scan direction, since the second ref
-        # file will never be used.
-        meta.subdata_ref.append(data.flux)
-        meta.diffmask_ref.append(meta.diffmask[-1])
-
-    print("Calculating 2D drift...")
+    log.writelog("  Calculating 2D drift...", mute=(not meta.verbose))
     # FINDME: instead of calculating scanHeight, consider fitting
     # stretch factor
-    drift2D = np.zeros((meta.nreads-1, 2))
-    meta.drift2D.append(drift2D)
+    drift2D = np.zeros((data.flux.shape[0], 2))
     if meta.ncpu == 1:
         # Only 1 CPU
-        # Get index of reference frame
-        # (0 = forward scan, 1 = reverse scan)
-        p = meta.scandir[m]
-        for n in range(meta.nreads-1):
-            writeDrift2D(hst.calcDrift2D((meta.subdata_ref[p][0] *
-                                          meta.subdiffmask[p][0]),
-                                         (data.flux[n] *
-                                          meta.subdiffmask[-1][n]),
-                                         m, n))
+        for n in range(data.flux.shape[0]):
+            # Get read number
+            r = n % meta.nreads
+            # Get index of reference frame
+            # (0 = forward scan, 1 = reverse scan)
+            p = data.scandir.values[n]
+            writeDrift2D(hst.calcDrift2D((meta.subdata_ref[p][r] *
+                                          meta.subdiffmask_ref[p][r]),
+                                         (data.flux[n]*data.flatmask[n]),
+                                         n))
     else:
         # Multiple CPUs
         pool = mp.Pool(meta.ncpu)
-        # Get index of reference frame
-        # (0 = forward scan, 1 = reverse scan)
-        p = meta.scandir[m]
-        for n in range(meta.nreads-1):
+        for n in range(data.flux.shape[0]):
+            # Get read number
+            r = n % meta.nreads
+            # Get index of reference frame
+            # (0 = forward scan, 1 = reverse scan)
+            p = data.scandir.values[n]
             res = pool.apply_async(hst.calcDrift2D,
-                                   args=((meta.subdata_ref[p][0] *
-                                          meta.subdiffmask[p][0]),
-                                         (data.flux[n] *
-                                          meta.subdiffmask[-1][n]),
-                                         m, n),
+                                   args=((meta.subdata_ref[p][r] *
+                                          meta.subdiffmask_ref[p][r]),
+                                         (data.flux[n]*data.flatmask[n]),
+                                         n),
                                    callback=writeDrift2D)
         pool.close()
         pool.join()
         res.wait()
 
-    print("Performing rough, pixel-scale drift correction...")
-    meta.drift2D_int.append(np.round(meta.drift2D[-1], 0))
+    data['drift2D'] = xr.DataArray(drift2D, name='drift2D',
```

Comment:
Instead, I think you could use `xrio.makeTimeLikeDA(drift2D, data.time, units='pixels', time_units=data.time.time_units, name='drift2D')`.

Reply:
Well that's annoying!  And `makeFluxLikeDA` requires flux_units as an input, which doesn't really make sense when talking about pixel shifts.  Guess I should have just used "units" in `makeFluxLikeDA`.